### PR TITLE
MRG, ENH: Link to methods and properties properly

### PR DESCRIPTION
**Pull request by larsoner**
Link to methods and properties properly (part of #587).

Locally I can confirm that building MNE docs now produces links to things like `ica.plot_properties`, which is great.

### CODE REVIEW

File: sphinx_gallery/backreferences.py [hunk 1]
```diff
@@ -23,6 +23,19 @@
...
+    """Dummy class for testing method resolution."""
+
+    def run(self):
+        """Do nothing."""
+        pass
+
+    @property
+    def prop(self):
+        """Property."""
+        return 'Property'
```
Comment (larsoner) [inline sphinx_gallery/backreferences.py]: This is used for `tinybuild` testing. The cost is a useless entry in our actual official docs but I don't think it's too bad.

File: sphinx_gallery/backreferences.py [hunk 2]
```diff
@@ -121,12 +125,20 @@ def get_mapping(self):
...
+                remainder = name[len(local_name):]
+                if local_name in self.imported_names:
                     full_name = self.imported_names[local_name] + remainder
-                    is_class = False
-                    yield name, full_name, class_attr, is_class
-                    break
+                    is_class = class_attr = False  # can't tell without import
+                    options.append(
+                        (name, full_name, class_attr, is_class))
+        return options
```
Comment (larsoner) [inline sphinx_gallery/backreferences.py]: @lucyleeow I simplified this a great deal. The diff is a bit ugly, but if you look at the actual file now hopefully it reads much more easily than it did before:

https://github.com/sphinx-gallery/sphinx-gallery/blob/58df0e6d472fab7e6745db6b01b9ee24333b3ab3/sphinx_gallery/backreferences.py#L80-L140

File: sphinx_gallery/tests/test_full.py [hunk 3]
```diff
@@ -191,7 +191,7 @@ def test_embed_links_and_styles(sphinx_app):
     assert 'pyplot.html' in lines
     assert 'matplotlib.figure.Figure.html#matplotlib.figure.Figure.tight_layout' in lines  # noqa: E501
     assert 'matplotlib.axes.Axes.plot.html#matplotlib.axes.Axes.plot' in lines
-    assert 'matplotlib_configuration_api.html#matplotlib.rcParams' in lines
+    assert 'matplotlib_configuration_api.html#matplotlib.RcParams' in lines
```
Comment (larsoner) [inline sphinx_gallery/tests/test_full.py]: The changes above modified how this was resolved, which I think is okay. The `rcParams` is an instance of `RcParams`, so this is still a legitimate documentation link to provide. (Whether it should choose one or the other will depend on the package itself, and we probably don't have a good way to know. Plus this should be rare, as it should be rare that packages have case-based naming differentiation in the same submodules, both of which get documented.)

**Comment by lucyleeow** [comment]
LGTM thanks @larsoner

**Pull request merged**